cartography/intel/aws/ec2/snapshots.py

```diff
@@ -12,13 +12,29 @@
 logger = logging.getLogger(__name__)
 
 
+@timeit
+def get_snapshots_in_use(neo4j_session: neo4j.Session, region: str, current_aws_account_id: str) -> List[str]:
+    query = """
+    MATCH (:AWSAccount{id: {AWS_ACCOUNT_ID}})-[:RESOURCE]->(v:EBSVolume)
+    WHERE v.region = {Region}
+    RETURN v.snapshotid as snapshot
+    """
+    results = neo4j_session.run(query, AWS_ACCOUNT_ID=current_aws_account_id, Region=region)
+    return [r['snapshot'] for r in results]
+
+
 @timeit
 @aws_handle_regions
-def get_snapshots(boto3_session: boto3.session.Session, region: str) -> List[Dict]:
+def get_snapshots(boto3_session: boto3.session.Session, region: str, snapshotIds: List[str]) -> List[Dict]:
     client = boto3_session.client('ec2', region_name=region)
     paginator = client.get_paginator('describe_snapshots')
     snapshots: List[Dict] = []
-    for page in paginator.paginate():
+    for page in paginator.paginate(OwnerIds=['self']):
+        snapshots.extend(page['Snapshots'])
+
+    if not snapshotIds:
+        return snapshots
+    for page in paginator.paginate(SnapshotIds=snapshotIds):
         snapshots.extend(page['Snapshots'])
     return snapshots
```

Comment:
Change `snapshotIds` to `in_use_snapshot_ids`.

Change L35-39 to something like
```
...
self_owned_snapshot_ids = set(s['SnapshitId'] for s in snapshots)

# reduce a a smaller list of unfetched snapshots
other_snapshot_ids = set(in_use_snapshot_ids) - self_owned_snapshot_ids
if other_snapshot_ids:
    for page in paginator.paginate(SnapshotIds=list(other_snapshot_ids)):
        snapshots.extend(page['Snapshots']
return snapshots
```

Reply:
👍 Updated!